src/devices/Uln2003/Uln2003.cs

```diff
@@ -0,0 +1,169 @@
+﻿using System;
+using System.Device.Gpio;
+using System.Diagnostics;
+
+namespace Iot.Device.Uln2003
+{
+    /// <summary>
+    /// This class is for controlling stepper motors that are controlled by a 4 pin controller board.
+    /// </summary>
+    /// <remarks>It is tested and developed using the 28BYJ-48 stepper motor and the ULN2003 driver board.</remarks>
+    public class Uln2003 : IDisposable
+    {
+        /// <summary>
+        /// Default delay in microseconds.
+        /// </summary>
+        private const long StepperMotorDefaultDelay = 1000;
+
+        private int _pin1;
+        private int _pin2;
+        private int _pin3;
+        private int _pin4;
+        private int _steps = 0;
+        private int _engineStep = 0;
+        private int _currentStep = 0;
+        private StepperMode _mode = StepperMode.HalfStep;
+        private bool[,] _currentSwitchingSequence = _halfStepSequence;
+        private bool _isClockwise = true;
+        private GpioController _controller;
+        private Stopwatch _stopwatch;
+        private long _stepMicrosecondsDelay;
+
+        static bool[,] _halfStepSequence = new bool[4, 8] {
```

Comment:
We do enforce all members in a class to use explicit access modifiers, so all these arrays should call out that they are private.